tests/basic/storage_pool_test.go

```diff
@@ -10005,3 +10006,341 @@ func exitPoolMaintenance(poolId string) {
+		Step(stepLog, func() {
+			// create spec to perform add drive
+			newSpecSize := poolToAddBack.TotalSize / units.GiB
+			driveSpecs, err := GetCloudDriveDeviceSpecs()
+			log.FailOnError(err, "Error getting cloud drive specs")
+			deviceSpec := driveSpecs[0]
+			deviceSpecParams := strings.Split(deviceSpec, ",")
+			paramsArr := make([]string, 0)
+			for _, param := range deviceSpecParams {
+				if strings.Contains(param, "size") {
+					paramsArr = append(paramsArr, fmt.Sprintf("size=%d,", newSpecSize))
+				} else {
+					paramsArr = append(paramsArr, param)
+				}
+			}
+			newSpec := strings.Join(paramsArr, ",")
+
+			err = Inst().V.AddCloudDrive(&selectedNode, newSpec, -1)
+			log.FailOnError(err, "error adding new drive to node %s", selectedNode.Name)
+			log.InfoD("Validate pool rebalance after drive add to the node %s", selectedNode.Name)
+			err = ValidateDriveRebalance(selectedNode)
+			log.FailOnError(err, "pool re-balance failed on node %s", selectedNode.Name)
+			err = Inst().V.WaitDriverUpOnNode(selectedNode, addDriveUpTimeOut)
+			log.FailOnError(err, "volume drive down on node %s", selectedNode.Name)
+
+			newPoolsMap, err := Inst().V.GetPoolDrives(&selectedNode)
+			log.FailOnError(err, "error getting pool drive from the node [%s]", selectedNode.Name)
+			dash.VerifyFatal(len(newPoolsMap) == 1, true, "verify new drive is created")
+		})
+
+		stepLog = fmt.Sprintf("Expand newly added pool on node [%s]", selectedNode.Name)
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			poolsAfr, err := Inst().V.ListStoragePools(metav1.LabelSelector{})
+			log.FailOnError(err, "Failed to list storage pools")
+			var poolIDSelected string
+			for k := range poolsAfr {
+				poolIDSelected = k
+				break
+			}
+			poolToBeResized, err := GetStoragePoolByUUID(poolIDSelected)
+			log.FailOnError(err, fmt.Sprintf("Failed to get pool using UUID %s", poolIDSelected))
+			expectedSize := (poolToBeResized.TotalSize / units.GiB) + 100
+
+			log.InfoD("Current Size of the pool %s is %d", poolIDSelected, poolToBeResized.TotalSize/units.GiB)
+			err = Inst().V.ExpandPool(poolIDSelected, api.SdkStoragePool_RESIZE_TYPE_AUTO, expectedSize, false)
+			dash.VerifyFatal(err, nil, "Pool expansion init successful?")
+
+			resizeErr := waitForPoolToBeResized(expectedSize, poolIDSelected, false)
+			dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Verify pool %s on expansion using auto option", poolIDSelected))
+		})
+
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts)
+	})
+})
+
+var _ = Describe("{PoolDeleteNegative}", func() {
+
+	/*
+				migrated from px-test: PoolDeleteNegative
+		 		1. Delete pool with invalid pool IDs
+				2. Delete pool with volumes on it
+				3. Delete pool after it was added to volume through ha-incr
+	*/
+
+	deletePoolAndValidateFaliure := func(poolID string, selectedNode *node.Node, errRegExp *regexp.Regexp) {
```

Comment:
have this inside it. this will give panic if it is not in It

Reply:
I didn't understand the comment.